couler/core/templates/container.py

```diff
@@ -193,9 +195,11 @@ def _convert_args_to_input_parameters(self, args):
                 o = args[i]
                 if isinstance(o, OutputArtifact):
                     para_name = o.artifact["name"]
-                    parameters.append('"{{inputs.artifacts.%s}}"' % para_name)
+                    param_full_name = '"{{inputs.artifacts.%s}}"' % para_name
                 else:
                     para_name = utils.input_parameter_name(self.name, i)
-                    parameters.append('"{{inputs.parameters.%s}}"' % para_name)
+                    param_full_name = '"{{inputs.parameters.%s}}"' % para_name
+                if param_full_name not in parameters:
```

Comment:
Move @javoweb's comment to the code comment too?

Reply:
Oh looks like this is the change for https://github.com/onepanelio/couler/pull/10? @rushtehrani

Reply:
> Oh looks like this is the change for [onepanelio#10](https://github.com/onepanelio/couler/pull/10)? @rushtehrani

@terrytangyuan no, that change removes artifacts from being injected into args, this just makes sure params with the same name aren't duplicated - I think adding @javoweb's comment here makes sense.